plugin.idea/src/com/microsoft/alm/plugin/idea/tfvc/ui/settings/EULADialog.java

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root.
+
+package com.microsoft.alm.plugin.idea.tfvc.ui.settings;
+
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.ui.DialogWrapper;
+import com.intellij.openapi.util.SystemInfo;
+import com.intellij.ui.components.JBScrollPane;
+import com.intellij.util.ui.JBUI;
+import com.microsoft.alm.helpers.Path;
+import com.microsoft.alm.plugin.external.tools.TfTool;
+import com.microsoft.alm.plugin.external.utils.ProcessHelper;
+import kotlin.text.Charsets;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.LoggerFactory;
+
+import javax.swing.*;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.Arrays;
+import java.util.jar.JarFile;
+
+public class EULADialog extends DialogWrapper {
+    private final static org.slf4j.Logger logger = LoggerFactory.getLogger(EULADialog.class);
+    private static boolean myWasShow = false;
+    private final JBScrollPane myScrollPane;
+    private final Boolean myEulaTextFound;
+
+    private EULADialog(@Nullable Project project) {
+        super(project);
+        String eulaText = getEULAText();
+        myEulaTextFound = eulaText != null;
+
+        JTextArea textArea = new JTextArea();
+        textArea.setText(eulaText);
+        textArea.setMinimumSize(null);
+        textArea.setEditable(false);
+        textArea.setCaretPosition(0);
+        textArea.setBackground(null);
+        textArea.setBorder(JBUI.Borders.empty());
+
+        myScrollPane = new JBScrollPane(textArea);
+        myScrollPane.setHorizontalScrollBarPolicy(JScrollPane.HORIZONTAL_SCROLLBAR_NEVER);
+
+        setModal(true);
+        setTitle("Team Foundation Command-Line Client EULA");
+
+        setOKButtonText("Accept");
+        getOKAction().putValue(DEFAULT_ACTION, null);
+
+        setCancelButtonText("Decline");
+        getCancelAction().putValue(DEFAULT_ACTION, Boolean.TRUE);
+
+        init();
+    }
+
+    @Nullable
+    private String getEULAText() {
+        try {
+            String tfLocation = TfTool.getLocation();
+            if (tfLocation == null) {
+                return null;
+            }
+            String path = Path.combine(Path.combine(Path.getDirectoryName(tfLocation), "lib"),
+                    "com.microsoft.tfs.client.common.jar");
+            JarFile jar = new JarFile(path);
+            InputStream stream = jar.getInputStream(jar.getEntry("eula.txt"));
+            BufferedReader rdr = new BufferedReader(new InputStreamReader(stream, Charsets.UTF_8));
```

Comment:
I'm not up-to-snuff on my Java, but we've opened a file here with `.getInputStream()`, do we need to explicitly close that file? Seems to me that we'd want to, but I'm not certain how Java handles these kinds of things.

Is it OK that we just wait for garbage collection to swing by and clean things up? Could waiting cause any kind of contention with the file being opened by multiple readers? (mostly thinking aloud here).